bindgen-cli/options.rs

```diff
@@ -1092,5 +1102,53 @@ where
         builder = builder.wrap_unsafe_ops(true);
     }
 
+    if let Some(custom_derives) = matches.values_of("with-derive-custom") {
+        #[derive(Debug)]
+        struct CustomDeriveCallback {
+            derive: String,
+            kind: String, 
+            regex: String,
+        }
+        impl bindgen::callbacks::ParseCallbacks for CustomDeriveCallback {
+            fn add_derives(&self, info: &bindgen::callbacks::DeriveInfo<'_>) -> Vec<String> {
+                match self.kind.to_lowercase().as_str() {
+                    "struct" => match info.kind {
+                        Some(bindgen::ir::comp::CompKind::Struct) => {}, // continue
```

Comment:
Hmm this brings up again a discussion we had with @emilio before. Should this `<kind>` be the kind of `c` types or `rust` types? For example some `c` `union`s are emitted as `rust` `struct`s and `CompKind` doesn't include `enum`s` as those aren't compound types per-se.

I'm more inclined to this based on the `rust` kinds instead as we're deriving `rust` traits on them but I'm open to hear other arguments.

Reply:
That's a good question... I think I agree with you... which kind of kind (😄 ) am I using here?

I was not expecting to not find enums in the enum for CompKind (this is starting to get meta...)
So I gave up on the branch of code that handles everything but structs and unions.

If this is referencing C types, how do I get the rust types to check?

Reply:
Deciding whether a comptype becomes a rust struct or union happens here: https://github.com/rust-lang/rust-bindgen/blob/9c2167c3bb6ca18822aca3c91e1b299bfc9d5338/bindgen/codegen/mod.rs#L2122-L2132